tests/test_publish_plugin_instance.py

```diff
@@ -145,3 +145,54 @@ def test_accept_with_accepted_item(self, create_hook_instance):
         handler.keyword = "Error running accept for"
         self.assertEqual(ppi.run_accept(None), {"accepted": True})
         self.assertFalse(handler.found)
+
+    # The following four methods are mock methods for the three UI customization methods.
+    # get_ui_settings
+    # set_ui_settings
+    # create_settings_widget
+    # We're not using proper return values, as we are only testing that the API handles
+    # passing the correct amount of arguments.
+
+    def _mock_get_create_w_items(self, parent, items):
+        return "passed"
+
+    def _mock_get_create_wo_items(self, parent):
+        return "passed"
+
+    def _mock_set_settings_w_items(self, parent, settings, items):
+        pass
+
+    def _mock_set_settings_wo_items(self, parent, settings):
+        pass
+
+    @mock_publish_plugin_instance_hook_creation
+    def test_plugin_ui_methods(self, create_hook_instance):
+        """
+        Ensure that the custom UI methods receive the correct number of arguments.
+        """
+
+        # Create hook instances for the UI methods that accept the items arg
+        create_hook_instance.return_value = MagicMock(
+            create_settings_widget=self._mock_get_create_w_items,
```

Comment:
I thought you had a bug here because you passed `_moc_create_w_items` to both `create_ui_settings` and `get_ui_settings`, until I realized they have both the same signature, right?

Maybe it would be less confusing if you simply used lambas here?